arangod/Cluster/Utils/ShardID.cpp

```diff
@@ -0,0 +1,83 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2023-2023 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Michael Hackstein
+////////////////////////////////////////////////////////////////////////////////
+
+#include "ShardID.h"
+
+#include "Basics/Exceptions.h"
+#include "Basics/StringUtils.h"
+
+#include <absl/strings/str_cat.h>
+
+arangodb::ResultT<arangodb::ShardID> arangodb::ShardID::shardIdFromString(
+    std::string_view s) noexcept {
```

Comment:
I would not declare this function `noexcept`, because in theory the creation of the `arangodb::Result` object can throw an exception in `std::make_unique<Error>()`.

Reply:
Done